src/test/java/org/gridsuite/ds/server/controller/DynamicSimulationControllerTest.java

```diff
@@ -124,144 +148,199 @@ public void setUp() throws IOException {
         assertEquals(runUuid, UUID.fromString(messageSwitch.getHeaders().get(HEADER_RESULT_UUID).toString()));
 
         //get the calculation status
-        EntityExchangeResult<DynamicSimulationStatus> entityExchangeResult2 = webTestClient.get()
-                .uri("/v1/results/{resultUuid}/status", runUuid)
-                .exchange()
-                .expectStatus().isOk()
-                .expectBody(DynamicSimulationStatus.class)
-                .returnResult();
+        result = mockMvc.perform(
+                get("/v1/results/{resultUuid}/status", runUuid))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        DynamicSimulationStatus status = objectMapper.readValue(result.getResponse().getContentAsString(), DynamicSimulationStatus.class);
 
         //depending on the execution speed it can be both
-        assertTrue(DynamicSimulationStatus.CONVERGED == entityExchangeResult2.getResponseBody()
-                || DynamicSimulationStatus.RUNNING == entityExchangeResult2.getResponseBody());
+        assertTrue(DynamicSimulationStatus.CONVERGED == status
+                || DynamicSimulationStatus.RUNNING == status);
 
         //get the status of a non-existing simulation and expect a not found
-        webTestClient.get()
-                .uri("/v1/results/{resultUuid}/status", UUID.randomUUID())
-                .exchange()
-                .expectStatus().isNotFound();
-
-        //get the results of a non-existing simulation and expect a not found
-        webTestClient.get()
-                .uri("/v1/results/{resultUuid}/timeseries", UUID.randomUUID())
-                .exchange()
-                .expectStatus().isNotFound();
-
-        //get the result timeseries uuid of the calculation
-        webTestClient.get()
-                .uri("/v1/results/{resultUuid}/timeseries", runUuid)
-                .exchange()
-                .expectStatus().isOk()
-                .expectBody(UUID.class);
+        mockMvc.perform(
+                get("/v1/results/{resultUuid}/status", UUID.randomUUID()))
+            .andExpect(status().isNotFound())
+            .andReturn();
+
+        //get the time-series uuid of a non-existing simulation and expect a not found
+        mockMvc.perform(
+                get("/v1/results/{resultUuid}/timeseries", UUID.randomUUID()))
+            .andExpect(status().isNotFound())
+            .andReturn();
+
+        //get the timeline uuid of a non-existing simulation and expect a not found
+        mockMvc.perform(
+                        get("/v1/results/{resultUuid}/timeline", UUID.randomUUID()))
+                .andExpect(status().isNotFound())
+                .andReturn();
+
+        //get the result time-series uuid of the calculation
+        mockMvc.perform(
+                get("/v1/results/{resultUuid}/timeseries", runUuid))
+            .andExpect(status().isOk())
+            .andReturn();
```

Comment:
no need to call "andReturn", same for next calls

Reply:
we've lost the "expectBody(UUID.class)", is it fine ?

Reply:
DONE!

Removed any andReturn if not needed

For checking whether a json string can be converted a class, I have created a utility method as following:

```
    public static void assertType(String response, Class<?> valueType, ObjectMapper objectMapper) {
        assertDoesNotThrow(() -> objectMapper.readValue(response, valueType));
    }
```